src/qt/trafficgraphwidget.cpp

```diff
@@ -46,27 +63,75 @@ int TrafficGraphWidget::getGraphRangeMins() const
     return nMins;
 }
 
-void TrafficGraphWidget::paintPath(QPainterPath &path, QQueue<float> &samples)
+//MLUMIN 05_22: Resampling function neccessary for re-fitting all acquired data samples into displayed visual sample canvas/view.
+float TrafficGraphWidget::resampleSamples(const QVector<float> &source, QVector<float> &dest) const
 {
-    int h = height() - YMARGIN * 2, w = width() - XMARGIN * 2;
-    int sampleCount = samples.size(), x = XMARGIN + w, y;
-    if(sampleCount > 0) {
-        path.moveTo(x, YMARGIN + h);
-        for(int i = 0; i < sampleCount; ++i) {
-            x = XMARGIN + w - w * i / DESIRED_SAMPLES;
-            y = YMARGIN + h - (int)(h * samples.at(i) / fMax);
-            path.lineTo(x, y);
+    float stride = (float)nSamplesWindowSize / dest.size();
+    float numSamplesToCombine = stride;
+    float maxSample = 0.0f;
+    float lastSample = 0.0f;
+
+    size_t currentSampleIndex = nSamplesTailIndex;
+    for (size_t destIndex = 0; destIndex < dest.size(); destIndex++) {
+        int numSamplesToRead = (int)numSamplesToCombine;
+
+        float currentSample = 0.0f;
+
+        for (size_t i = 0; i < numSamplesToRead; ++i) {
+            currentSample = std::max(currentSample, source[currentSampleIndex]);
```

Comment:
So, actually, I began with averaging - and it really looked wrong; and i do think that's because of the aliasing effect of resampling basically 'double smoothing'. I'll show what this looked like in a bit, its a pretty easy change; but in effect I think the act of resampling may be doing averaging inherently. In signals speak if you average and then resample you're killing transient response time.  

-- What I wanted to do, and, maybe propose to do? Is take the root mean square (RMS) value. I think that will actually be more visually appropriate, in that it will not attenuate outlying transients in ampltude (basically the effect of a spike will still be displayed), while time domain occurance and spread (transients in frequency) will be taken into account.  Less performant though. 

rms(1,1,101,1,1) = 45 .

Will show what avg looked like here though in a bit.

Reply:
Ok so sped up - RMS didn't behave as nice as I would have hoped either. But this is avg (sped up for visibility):




https://user-images.githubusercontent.com/6802127/167223787-ecad0c5a-98f6-4978-991f-07e37d848397.mp4



But notice how with average (at this point); two things happen:

1) its very jittery and 'dancing', due to new averages being taken from different windows of data, 
2) when at max time it basically averages everything int a big (small/low biased) lump.

I'm looking at this - this may have to be a 'sliding window' averaging before the decimation happens. 

-- Give me a little bit to see if I can get this behaving better under an averaged reading.